### PR TITLE
AOD: Fix thinner for tracked 2023pp

### PR DESCRIPTION
**Pull request by f3sch**
Hi @jgrosseo, @ddobrigk,

this is a reduced version of https://github.com/f3sch/AliceO2/tree/thinner/fix (the output of both branches is identical, the latter is IMO clearer). With this we keep the full trackQA table. I ran a strangeness cascade analysis on the output which reproduces identical results as on the original AO2D. I would still suggest that a large test should be conducted.

### CODE REVIEW

File: Framework/AODMerger/src/aodMerger.h [hunk 1]
```diff
@@ -10,14 +10,18 @@
 // or submit itself to any jurisdiction.
 
 #include <TString.h>
+#include <TRegexp.h>
 
 const char* removeVersionSuffix(const char* treeName)
 {
   // remove version suffix, e.g. O2v0_001 becomes O2v0
+  // handle O2track_iu correctly
```
Comment (jgrosseo) [inline Framework/AODMerger/src/aodMerger.h]: So track_iu becomes track. And wasn't this an unintended but needed feature?

E.g. AmbiguousTracks has a column TrackId which in the AO2D needs to have the offset re-indexed (in https://github.com/AliceO2Group/AliceO2/blob/024bf20093dbe75d501a1670f60eeb8699645b80/Framework/AODMerger/src/aodMerger.cxx#L377) of track_iu. However, the column is called fTrackId which would not be found in the new version anymore.

Can you try to run the merged with this change? It should complain about unbound indices...

File: Framework/AODMerger/src/aodThinner.cxx [hunk 2]
```diff
@@ -194,22 +200,38 @@ int main(int argc, char* argv[])
...
+    }
 
     // We need to loop over the V0s once and flag the prong indices
     int trackIdxPos = 0, trackIdxNeg = 0;
-    std::unordered_map<int, bool> keepV0TPCs;
+    std::vector<bool> keepV0s(trackExtraTree->GetEntries(), false);
     v0s->SetBranchAddress("fIndexTracks_Pos", &trackIdxPos);
     v0s->SetBranchAddress("fIndexTracks_Neg", &trackIdxNeg);
-    auto nV0s = v0s->GetEntriesFast();
+    auto nV0s = v0s->GetEntries();
```
Comment (jgrosseo) [inline Framework/AODMerger/src/aodThinner.cxx]: Why this change (also elsewhere...)?

File: Framework/AODMerger/src/aodThinner.cxx [hunk 3]
```diff
@@ -194,22 +200,38 @@ int main(int argc, char* argv[])
...
-      keepV0TPCs[trackIdxPos] = true;
-      keepV0TPCs[trackIdxNeg] = true;
+      keepV0s[trackIdxPos] = true;
+      keepV0s[trackIdxNeg] = true;
+    }
+
+    // TrackQA
+    std::vector<bool> keepTrackQA;
+    if (hasTrackQA) {
+      keepTrackQA = std::vector<bool>(trackExtraTree->GetEntries(), false);
```
Comment (jgrosseo) [inline Framework/AODMerger/src/aodThinner.cxx]: This doesn't copy?

File: Framework/AODMerger/src/aodThinner.cxx [hunk 4]
```diff
@@ -367,15 +384,6 @@ int main(int argc, char* argv[])
...
 
-        // Reassign v0 index of cascades
-        if (processingCascades) {
-          if (keepV0s[indexV0s] < 0) {
-            fillThisEntry = false;
-          } else {
-            indexV0s = keepV0s[indexV0s];
-          }
-        }
-
```
Comment (jgrosseo) [inline Framework/AODMerger/src/aodThinner.cxx:378]: So this part is gone because we keep all V0s and cascades, right?

**Review by jgrosseo — COMMENTED** [review]
Thanks for this work! Please find a few comments/questions inline.

File: Framework/AODMerger/src/aodMerger.h (hunk 1 above)
Comment (f3sch) [reply Framework/AODMerger/src/aodMerger.h]: You are right. I will drop this change and with it the sanity-check at the end.

File: Framework/AODMerger/src/aodThinner.cxx (hunk 2 above)
Comment (f3sch) [reply Framework/AODMerger/src/aodThinner.cxx]: Remnant of me thinking first lets use `GetEntriesFast` its faster, then realising that `GetEnries` and `GetEntriesFast` are equivalent in newer ROOT version and changing everything to the same. If not wanted can drop it?

File: Framework/AODMerger/src/aodThinner.cxx (hunk 4 above)
Comment (f3sch) [reply Framework/AODMerger/src/aodThinner.cxx:378]: yes

**Comment by miranov25** [comment]
I confirm the code is doing what it is supposed to do in respect to the trackQA table as demonstrated  in Felix QA plot:
[trackqa.pdf](https://github.com/user-attachments/files/16105873/trackqa.pdf)

**Review by jgrosseo — APPROVED** [review]

**Comment by jgrosseo** [comment]
OK, I merged this. 
@chiarazampolli I think we should do right away a test on one of the 2023 periods to validate this on a larger sample
